core/trino-main/src/main/java/io/trino/sql/planner/Symbol.java

```diff
@@ -14,46 +14,59 @@
 package io.trino.sql.planner;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import io.trino.spi.type.Type;
 import io.trino.sql.ir.Expression;
 import io.trino.sql.ir.SymbolReference;
 
-import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+@JsonSerialize(keyUsing = SymbolKeySerializer.class)
 public class Symbol
         implements Comparable<Symbol>
 {
     private final String name;
+    private final Type type;
 
     public static Symbol from(Expression expression)
     {
-        checkArgument(expression instanceof SymbolReference, "Unexpected expression: %s", expression);
-        return new Symbol(((SymbolReference) expression).getName());
+        return switch (expression) {
+            case SymbolReference symbol -> new Symbol(symbol.type(), symbol.name());
+            default -> throw new IllegalArgumentException("Unexpected expression: " + expression);
```

Comment:
Is the plan that we will have more of these?  If not, I have been preferring `if (expression instance of SymbolReference symbol)` instead of a single case switch

Reply:
No, just that one. Updated.